readable-regexes/src/pydis_jam23/web/__init__.py

```diff
@@ -0,0 +1,122 @@
+import io
+import webbrowser
+from typing import Any
+
+from flask import Flask, flash, render_template, request, send_file
+from PIL import Image
+
+from pydis_jam23.codecs import CODECS, Codec, CodecError, CodecParam
+
+app = Flask(__name__)
+app.secret_key = "secret"  # We're only running locally.
+
+# Yep, it's a global mutable variable. It *should* work fine though since
+# there's only one user.
+current_image: Image.Image | None = None
+
+
+def run_server():
+    webbrowser.open("http://localhost:5000", new=2)
+    app.run()
+
+
+@app.route("/")
+def index():
+    return render("index.j2")
+
+
+@app.route("/codec/<codec>")
+def get_codec(codec: str):
+    return render("codec.j2", codec=find_codec(codec))
+
+
+@app.route("/encode/<codec>", methods=["GET"])
+def get_encode(codec: str):
+    return render("action.j2", codec=find_codec(codec), encode=True)
+
+
+@app.route("/encode/<codec>", methods=["POST"])
+def post_encode(codec: str):
+    global current_image  # noqa: PLW0603 - global mutable
+    if current_image is None:
+        msg = "No image loaded"
+        raise ValueError(msg)
```

Comment:
I don't think it's valuable to store the message in a separate variable here. It adds an additional line for no particular reason.